src/js/Search/SearchVideoWrapper.js

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable no-param-reassign */
+import { CLASSNAME, NUMBER, SNACKBAR_TEXT } from "../constants/index.js";
+import { messenger, MESSAGE } from "../messenger/index.js";
+import { renderSearchVideo, SEARCH_VIDEO_TEMPLATE } from "../Video/index.js";
+import { $, fetchYoutubeData, showModalSnackbar } from "../utils/index.js";
+
+export default class SearchVideoWrapper {
+  constructor() {
+    this.initializeVariables();
+    this.selectHTMLElements();
+    this.addMessageListeners();
+    this.addEventListeners();
+  }
+
+  initializeVariables() {
+    this.currentQuery = "";
+    this.currentNextPageToken = "";
+    this.videoItemsMap = new Map();
+  }
+
+  selectHTMLElements() {
+    this.$searchVideoWrapper = $(`.${CLASSNAME.SEARCH_VIDEO_WRAPPER}`);
+    this.$notFoundImg = $(`.${CLASSNAME.NOT_FOUND_IMAGE}`);
+  }
+
+  addMessageListeners() {
+    messenger.addMessageListener(MESSAGE.KEYWORD_SUBMITTED, ({ query }) => {
+      this.currentQuery = query;
+
+      $.hide(this.$notFoundImg);
+      $.clear(this.$searchVideoWrapper);
+
+      this.mountTemplate();
+      this.$searchVideoWrapper.scrollTo({ top: 0 });
+    });
+
+    messenger.addMessageListener(
+      MESSAGE.DATA_LOADED,
+      ({ nextPageToken, items }) => {
+        if (items.length === 0) {
+          $.clear(this.$searchVideoWrapper);
+          $.show(this.$notFoundImg);
+
+          return;
+        }
+
+        this.currentNextPageToken = nextPageToken;
+
+        this.attachData(items);
+      }
+    );
+
+    messenger.addMessageListener(MESSAGE.SAVED_VIDEO_DELETED, ({ videoId }) => {
+      const $button = $(
+        `.${CLASSNAME.SAVE_VIDEO_BUTTON}[data-video-id="${videoId}"]`
+      );
+
+      this.setButtonSaveMode($button);
+    });
+  }
+
+  addEventListeners() {
+    this.$searchVideoWrapper.addEventListener(
+      "scroll",
+      this.handlePageScroll.bind(this)
+    );
+
+    this.$searchVideoWrapper.addEventListener(
+      "click",
+      ({ target: $target }) => {
```

Comment:
```suggestion
      (event) => {
      const $target = event.target;
```

이렇게 적는게 가독성이 더 좋은 것 같아요

Reply:
해당 부분은 리팩토링으로 인해 제거되었습니다.